src/services/apiService.ts

```diff
@@ -0,0 +1,22 @@
+import api from "@/services/api";
+
+export default {
+  async getHex(hex_string: string) {
+    const response = await api.get(`hex/${hex_string}`);
+    return response.data;
+  },
+  async getInfo() {
+    const response = await api.get(`info`);
+    return response.data;
+  },
+  async getScapy(protocols: Array<string>) {
+    return await api
+      .post(`packets`, { packets: protocols })
+      .then((response) => response.data["packets"]);
+  },
+  async createPacket(packetInfo: Array<any>) {
```

Comment:
creating types for data that is being sent to api can be tricky. My usual approach is to console log the getter api (something like api.getPacket) or response object and construct an interface based on that